--- a/TypeContext/TypeContextFactory.php
+++ b/TypeContext/TypeContextFactory.php
@@ -249,7 +249,7 @@ class TypeContextFactory
             try {
                 if (null !== $typeString) {
                     $type = $this->stringTypeResolver->resolve($typeString, $typeContext);
                 }
-            } catch (UnsupportedException) {
+            } catch (UnsupportedException|\InvalidArgumentException) {
             }

             $templates[$tag->value->name] = $type;
